src/components/players/list/index.js

```diff
@@ -12,51 +14,83 @@ class List extends React.Component {
     constructor(props) {
         super(props);
 
-        this.ref = createRef();
-
         this.state = {
-            selectedId: null,
-            selectedTeam: this.props.selectedTeam,
+            selectedPlayerId: null,
         };
+
+        this.ref = createRef();
     };
 
     componentDidUpdate(prevProps) {
-        if (prevProps.selectedTeam !== this.props.selectedTeam) {
-            this.setState({ selectedId: null });
-            this.ref.current.scrollTo(0, 0);
+        // console.log(this.props.selectedTeam.isLoading);
+
+        // if (this.props.selectedTeam.isLoading) {
+        //     // this.ref.current.scrollTo(0,0);
+        //     this.ref.current.style.overflow = 'hidden';
+        // } else {
+        //     this.ref.current.style.overflow = 'auto';
+        // }
+
+        if (prevProps.selectedTeam !== this.props.selectedTeam) { 
+            this.setState({ selectedPlayerId: null });
+            // this.ref.current.scrollTo(0,0);
         };
     };
 
     renderPlayers() {
-        const { players, selectedTeam } = this.props;
+        const { 
+            players, 
+            selectedTeam: {
+                isLoading,
+                team: { teamId }
+            } 
+        } = this.props;
 
-        return players.map((player, index) => {
-            const isSelected = this.state.selectedId === player.person_id;
-            const { teamColor } = player;
+        // const { isLoading, team } = selectedTeam; 
+        // const { teamId } = team;
+        // console.log('isLoading renderPlayers', isLoading);
 
+        return players.map((player, index) => {
```

Comment:
Here's the trick:

Set empty object arrays when isLoading is true:

```
        const emptyPlayers = new Array(20).fill({});

        return (isLoading ? emptyPlayers : players).map((player, index) => {
           ...
```

Now this will always reset the players list to empty object and placeholders

Reply:
Conditional concatenation, thx for the tip.